src/libsystemd/sd-device/device-monitor.c

```diff
@@ -627,8 +627,9 @@ _public_ int sd_device_monitor_filter_update(sd_device_monitor *m) {
         if (m->filter_uptodate)
                 return 0;
 
-        if (hashmap_isempty(m->subsystem_filter) &&
-            set_isempty(m->tag_filter)) {
+        if (m->snl.nl.nl_groups == MONITOR_GROUP_KERNEL || (
+                            hashmap_isempty(m->subsystem_filter) &&
+                            set_isempty(m->tag_filter))) {
```

Comment:
Indentation is spurious.

Reply:
This how emacs wants it to be (with systemd's `.dir-locals.el` in place and active). Anyway, I'll retry.